core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/recipe/CcmKeyDeregisterHandler.java

```diff
@@ -0,0 +1,66 @@
+package com.sequenceiq.cloudbreak.reactor.handler.recipe;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.ccm.termination.CcmResourceTerminationListener;
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.reactor.api.event.StackFailureEvent;
+import com.sequenceiq.cloudbreak.reactor.api.event.recipe.CcmKeyDeregisterRequest;
+import com.sequenceiq.cloudbreak.reactor.api.event.recipe.CcmKeyDeregisterSuccess;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.flow.event.EventSelectorUtil;
+import com.sequenceiq.flow.reactor.api.handler.EventHandler;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class CcmKeyDeregisterHandler implements EventHandler<CcmKeyDeregisterRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CcmKeyDeregisterHandler.class);
+
+    @Inject
+    private EventBus eventBus;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private CcmResourceTerminationListener ccmResourceTerminationListener;
+
+    @Override
+    public void accept(Event<CcmKeyDeregisterRequest> requestEvent) {
+        CcmKeyDeregisterRequest request = requestEvent.getData();
+        Selectable result;
+        try {
+            Stack stack = stackService.getByIdWithListsInTransaction(request.getResourceId());
+            if (Boolean.TRUE.equals(request.getUseCcm())) {
+                LOGGER.debug("De-registering key from CCM. Cluster CRN: {}",
+                        stack.getResourceCrn());
+                try {
+                    ccmResourceTerminationListener.deregisterCcmSshTunnelingKey(request.getActorCrn(), request.getAccountId(), request.getKeyId());
+                } catch (Exception ex) {
+                    LOGGER.warn("CCM key deregister failed", ex);
+                }
+            } else {
+                LOGGER.info("CCM is DISABLED, skipping de-registering of key from CCM. Cluster CRN: {}",
+                        stack.getResourceCrn());
+            }
+            result = new CcmKeyDeregisterSuccess(stack.getId());
+        } catch (Exception ex) {
+            LOGGER.warn("CCM key deregister failed", ex);
+            result = new StackFailureEvent(request.getResourceId(), ex);
```

Comment:
This block is weird because any exception thrown from the deregistration attempt is caught in a prior catch block. I guess this one only triggers if stackService.getByIdWithListsInTransaction fails, in which case the log message is misleading - it didn't really even try to deregister.

If this block stays, the logging should be at error level, since it leads to a StackFailureEvent.

Reply:
Will change the error message and log level.